### DISCUSSION TITLE
Need Guidance on Using Nav2 for Navigation with ROS2 Humble for Go2 robots

### DISCUSSION THREAD

**Discussion by rawias** [abizovnuralem/go2_omniverse#17] [Q&A]
I am currently using ROS2 Humble and Isaac Sim 4.0.0 to work with Go2 robots. According to the README file of this repository, Nav2 navigation is supported as indicated by the checklist item "9. Nav2 with Slam_toolbox ✅". However, after running `./run_sim.sh`, I found no topics to assign goal positions to the robots. As explained in the instructions, the only available control method is using the WASD keys.

Additionally, in another repository [go2_ros2_sdk](https://github.com/abizovnuralem/go2_ros2_sdk), which is also mentioned in the README file that it can be the base for the ROS2 setup, it says that "Navigation (nav2) (in the current version is not working, need to fix params)". 

I would like to request guidance on how I can enable and use Nav2 for navigation with Go2 robots. Specifically, I am looking for some instructions or examples on how to set goal positions via Nav2 and send `/cmd_vel` for each of my multiple robots following the setup explained in this repo.

Your assistance would be greatly appreciated. 
Thank you for your help in advance

**Comment by abizovnuralem** [comment]
Hello @rawias , currently go2_ros2_sdk nav2 support is on hold. I need some time to deal with that. If you want, you can help with that. The nav2 support was on the previous commits of this repo and go2_ros2_sdk, you can see the examples. Before multi robot support commits,

**Reply by rawias** [reply]
Hello @abizovnuralem , I am sincerely not sure which commits of this repo you are referring to. Would you be able to point me to the specific commits of this repo and also [go2_ros2_sdk](https://github.com/abizovnuralem/go2_ros2_sdk), where the Nav2 support was working? 

Thank you so much in advance

**Reply by abizovnuralem** [reply]
The latest repo is now working for nav2, you can try it

**Comment by delenius** [comment]
Just found this discussion/issue after trying to get this (nav2+slam_toolbox) to work (without looking at the existing code) for some time...
Some issues I ran into (I should note that I am not a ROS expert):
- The robot does not publish transforms from map to odom, or base_link to the laser/lidar. (I added a static transform for the former, not sure how important the latter is).
- The cmd_vel API is not what nav2 expects. Normally, the robot should stop when no more messages are received. This one keeps moving. I added a node to send a "stop" message if no cmd_vel message has been sent for 0.5 seconds.
- point_cloud2 messages do not include a time stamp (easy to add in ros2.py).
- I was not able to sync camera or point cloud with other topics (see issue #19 )

Current issue: slam_toolbox is not publishing any messages on /map, and has this error:
`[robot0.slam_toolbox]: Failed to compute odom pose`

I would be happy to help with this, if I can.

**Reply by abizovnuralem** [reply]
Good job, @delenius ! Have you tried the latest update for go2_ros2_sdk?
